### PR TITLE
mostly new NWM routelink

### PR DESCRIPTION
**Pull request by jmccreight**

### CODE REVIEW

**Comment by jmccreight** [comment]
There are a few nicknacks in here to observe

File: tests/config_file_meta/CONUS/nwm_ana/hydro.namelist [hunk 1]
```diff
@@ -18,24 +18,24 @@
     hydrotbl_f = './NWM/DOMAIN/hydro2dtbl_FullRouting_NWMv2.0.nc'
     igrid = 1
     io_config_outputs = 0
-    io_form_outputs = 2
+    io_form_outputs = 4
```
Comment (kafitzgerald) [inline tests/config_file_meta/CONUS/nwm_ana/hydro.namelist]: I believe we set this to 2 in operations.  Is this just for testing?

File: tests/config_file_meta/CONUS/nwm_ana/hydro.namelist [hunk 2]
```diff
@@ -18,24 +18,24 @@
...
     land_spatial_meta_flnm = './NWM/DOMAIN/GEOGRID_LDASOUT_Spatial_Metadata_1km_NWMv2.0.nc'
     lsmout_domain = 1
     nsoil = 4
     order_to_write = 1
     out_dt = 60
     outlake = 1
-    output_channelbucket_influx = 2
+    output_channelbucket_influx = 0
```
Comment (kafitzgerald) [inline tests/config_file_meta/CONUS/nwm_ana/hydro.namelist]: Is there a reason to turn these off?

Comment (kafitzgerald) [reply tests/config_file_meta/CONUS/nwm_ana/hydro.namelist]: Hawaii now appears to be inconsistent w/ CONUS

File: tests/config_file_meta/CONUS/nwm_ana/hydro.namelist (hunk 1 above)
Comment (jmccreight) [reply tests/config_file_meta/CONUS/nwm_ana/hydro.namelist]: These are the testing namelists. Not operations. 
This change is required for perfect restarts on channel-only. I guess there are other sneakier ways to do this, but this is the clearest. 
We should eventually have tests for the operational outputs and such, but the current namelists focus on the engineering tests.

File: tests/config_file_meta/CONUS/nwm_ana/hydro.namelist (hunk 2 above)
Comment (jmccreight) [reply tests/config_file_meta/CONUS/nwm_ana/hydro.namelist]: ... I guess we are doing the sneakier way here but not above. I will look at HI.  This is a good point.

**Comment by jmccreight** [comment]
OK. I believe we should have
io_form_outputs = 4 
everywhere for sanity that we are doing engineering tests on output with full floating precision. 

I will made the 
output_channelbucket_influx = 0
set only in base, so this will turn it off for HI in testing. Channel-only tests on HI will turn it on within the test.

**Comment by jmccreight** [comment]
Joe made a comment which I logged as an issue #204

**Pull request merged**